### DISCUSSION TITLE
Factor CLI abstractions out into module, for use by e.g., webUI

### DISCUSSION THREAD

**Discussion by squaremo** [fluxcd/flux2#707] [Proposals] [answered]
The webUI will end up recapitulating many of the operations used by the CLI; for instance, https://github.com/fluxcd/webui/pull/8.

It would be useful to separate those operations out into their own module, so they can be reused instead of duplicated. This will surely also help third-party integrations, e.g., further developments from https://github.com/weaveworks/eksctl/pull/3066.

**Comment by stefanprodan** [comment, marked as answer]
Relates to https://github.com/fluxcd/flux2/issues/291
